h2o-core/src/main/java/water/fvec/NewChunk.java

```diff
@@ -802,7 +802,7 @@ private void append2slowstr() {
         int nzs = 0; // assume one non-null for the element currently being stored
         for( int i:_is) if( i != -1 ) ++nzs;
         if( (nzs+1)*_sparseRatio < _len)
-          set_sparse(nzs, Compress.ZERO);
+          set_sparse(nzs, Compress.NA);
```

Comment:
This branch of execution is for String values (_id == null & _is != null). It doesn't make sense to use ZERO-compression because there will not be any numerical values. If ZERO compression is used, the code in type() method will incorrectly determine number of NAs (it will consider sparse NAs to actual be sparse ZEROs). For empty String chunks this will switch the chunk type to Numeric instead of String.